spec/controllers/physical_rack_controller_spec.rb

```diff
@@ -0,0 +1,46 @@
+describe PhysicalRackController do
+  render_views
+
+  let(:physical_rack) do
+    ems = FactoryGirl.create(:ems_physical_infra)
+    FactoryGirl.create(:physical_rack, :ems_id => ems.id, :id => 1)
+  end
+
+  before do
+    stub_user(:features => :all)
+    EvmSpecHelper.local_miq_server(:zone => FactoryGirl.build(:zone))
+    EvmSpecHelper.create_guid_miq_server_zone
+    login_as FactoryGirl.create(:user)
+  end
+
+  describe "#show_list" do
+    subject { get(:show_list) }
+
+    it "renders a GTL page" do
+      is_expected.to have_http_status 200
+      is_expected.to render_template(:partial => "layouts/_gtl")
```

Comment:
you can add a test for the /report_data JSON ednpoint to test what data is returned for the GTL:
https://github.com/ManageIQ/manageiq-ui-classic/pull/2830/files#diff-a26345b9b9b90896848419066359f6bfR225

and you can also add a test for the GTL -- making sure that it's rendered with the correct parameters
https://github.com/ManageIQ/manageiq-ui-classic/pull/2830/files#diff-a26345b9b9b90896848419066359f6bfR212

then it's quite easy to test if the proper toolbar is rendered:
https://github.com/ManageIQ/manageiq-ui-classic/pull/3857/files#diff-a32fdcf11645e7b22301a840fd2d6d7eR285

Reply:
Thansk @martinpovolny , didn't add tests for the toolbar tho, I plan to this in another PR that actually includes the toolbar